File: examples/wasm/js/wasm.js
var memory;

const getRandomSeed = () => Math.floor(Math.random() * 2147483647);
const getRandomString = () => Math.random().toString(36).substring(5, 15) + Math.random().toString(36).substring(5, 15);
const consoleLog = (value) => console.log('consoleLog', value);
const readCharStr = (ptr, len) => {
    const bytes = new Uint8Array(memory.buffer, ptr, len);
    let s = "";
    for (let i = 0; i < len; ++i) {
        s += String.fromCharCode(bytes[i]);
    }
    return s;
}

var wasm = {
    consoleLog,
    getRandomSeed,
    getRandomString,
    consoleLogF: consoleLog,
    consoleLogS: consoleLog,
    readCharStr,
}

File: examples/wasm/platform/web/wasm.zig

pub extern fn getRandomSeed() c_uint;
pub extern fn getRandomString() [*]const u8;
pub extern fn consoleLog(_: c_int) void;
pub extern fn consoleLogF(_: f32) void;
pub extern fn consoleLogS(_: [*]const u8, _: c_uint) void;